apps/omg/test/omg/conformance/merkle_proof_test.exs

```diff
@@ -0,0 +1,81 @@
+# Copyright 2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Conformance.MerkleProofTest do
+  @moduledoc """
+  Checks if some particular cases of merkle proofs (proof generation and validation) behave consistently across
+  implementations (currently `elixir-omg` and `plasma-contracts`, Elixir and Solidity)
+  """
+
+  alias OMG.Merkle
+  alias Support.SnapshotContracts
+
+  import Support.Conformance.MerkleProofs, only: [solidity_proof_valid: 5]
+
+  use ExUnit.Case, async: false
+
+  @moduletag :integration
+  @moduletag :common
+
+  @proof_length 16
+  @max_block_size trunc(:math.pow(2, @proof_length))
+
+  setup_all do
+    {:ok, exit_fn} = Support.DevNode.start()
+
+    contracts = SnapshotContracts.parse_contracts()
+    merkle_wrapper_address_hex = contracts["CONTRACT_ADDRESS_MERKLE_WRAPPER"]
+
+    on_exit(exit_fn)
+
+    [contract: OMG.Eth.Encoding.from_hex(merkle_wrapper_address_hex)]
+  end
+
+  test "a simple, 3-leaf merkle proof validates fine", %{contract: contract} do
+    leaves = [<<1>>, <<0>>, <<>>]
+    root_hash = Merkle.hash(leaves)
+
+    leaves
+    |> Enum.with_index()
+    |> Enum.each(fn {leaf, txindex} ->
+      proof = Merkle.create_tx_proof(leaves, txindex)
+      assert solidity_proof_valid(leaf, txindex, root_hash, proof, contract)
+    end)
+  end
+
+  @tag timeout: 240_000
+  test "a full-tree merkle proof validates fine", %{contract: contract} do
```

Comment:
i'm not 100% sure I want this. I want this tested, but (see comments) this needs to be slightly inconsistent to be practical.

Reply:
I feel the pain. I would delete this test, but I have no other reasons than that it tests `MerkleTree` lib itself and doesn't even touch `OMG.Merkle`.

Reply:
note that `OMG.Merkle` is a thin wrapper (really just limits the API and sets some flags) around `MerkleTree`. So testing on a "lower level" still kinda tests what is supposed to be tested, it's just a tiny bit more brittle and inconsistent. Hm... :thinking: 

OTOH, leaving the 2^16 with no test anywhere makes me feel uneasy a bit. It is a valid edge case.

Reply:
I'll leave it for now. These tests are planned to be moved over to `ex_plasma` and possibly leverage a different Merkling API, so maybe the problem will fix itself then.